### PR TITLE
enhancement(elasticsearch sink): Add update support to bulk action

### PR DESCRIPTION
**Pull request by blackrez**
## Summary
Like discussed in https://github.com/vectordotdev/vector/issues/17852 I add the update support to the bulk API. We deployed it and it works like we wanted.

## Change Type
- [ ] Bug fix
- [x] New feature
- [ ] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?
- [ ] Yes
- [x] No

## How did you test this PR?
We use Elasticsearch with the following config files :
https://gist.github.com/blackrez/8b5ff94e081fe68c7811deb8c9e44079 

## Does this PR include user facing changes?

- [X] Yes. Please add a changelog fragment based on our [guidelines] 
- [ ] No. A maintainer will apply the "no-changelog" label to this PR.

## Checklist
- [X] Please read our [Vector contributor resources](https://github.com/vectordotdev/vector/tree/master/docs#getting-started).

## References

- Closes: https://github.com/vectordotdev/vector/issues/17852

### CODE REVIEW

**Review by pront — APPROVED** [review]
Hi @blackrez, thank you for the PR. This looks straightforward.

File: website/cue/reference/components/sinks/base/elasticsearch.cue [hunk 1]
```diff
@@ -249,7 +249,7 @@ base: components: sinks: elasticsearch: configuration: {
 				description: """
 					Action to use when making requests to the [Elasticsearch Bulk API][es_bulk].
 
-					Only `index` and `create` actions are supported.
+					Only `index`, `create` and `update` actions are supported.
```
Comment (drichards-87) [inline website/cue/reference/components/sinks/base/elasticsearch.cue:252]:
```suggestion
					Only `index`, `create`, and `update` actions are supported.
```

File: website/cue/reference/components/sinks/elasticsearch.cue [hunk 2]
```diff
@@ -91,6 +91,9 @@ components: sinks: elasticsearch: {
 				inserted via the `index` action, which replaces documents if an existing
 				one has the same `id`. If `bulk.action` is configured with `create`, Elasticsearch
 				does _not_ replace an existing document and instead returns a conflict error.
+				When `bulk.action` is set to `update`, the document will be updated with several constraints.
```
Comment (drichards-87) [inline website/cue/reference/components/sinks/elasticsearch.cue:94]:
```suggestion
				When `bulk.action` is set to `update`, the document is updated with several constraints.
```

File: website/cue/reference/components/sinks/elasticsearch.cue [hunk 3]
```diff
@@ -91,6 +91,9 @@ components: sinks: elasticsearch: {
...
+				The message must be added in `.doc` and have `.doc_as_upsert` to true.
+				The `update` operation needs `id_key` set. The `encoding` field must precise `doc` and `doc_as_upsert` as values.
```
Comment (drichards-87) [inline website/cue/reference/components/sinks/elasticsearch.cue:96]:
```suggestion
				The `update` operation requires the `id_key` to be set, and the `encoding` field should specify `doc` and `doc_as_upsert` as values.
```

**Review by drichards-87 — APPROVED** [review]
Left a few minor suggestions from Docs and approved the PR.

**Pull request merged**